### ISSUE TITLE
MappedFile slow tailer is blocked by appender calling raf.setLength()

### ISSUE THREAD

**Issue by RobAustin** [OpenHFT/Chronicle-Bytes#120]
A tailer which is many thousand messages behind is blocked by an appender, I feel the solution here is to make the tailer acquireByteStore as readOnly for tailers

tailer:
```
"tree-1/core-event-loop" #13 daemon prio=5 os_prio=31 tid=0x00007ff0671af000 nid=0x4503 waiting for monitor entry [0x000070000b20a000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at net.openhft.chronicle.bytes.MappedFile.acquireByteStore(MappedFile.java)
	- waiting to lock <0x000000074603b310> (a net.openhft.chronicle.bytes.MappedFile)
	at net.openhft.chronicle.bytes.MappedFile.acquireByteStore(MappedFile.java:310)
	at net.openhft.chronicle.bytes.MappedBytes.acquireNextByteStore0(MappedBytes.java:434)
	- locked <0x000000074601cac0> (a net.openhft.chronicle.bytes.MappedBytes)
	at net.openhft.chronicle.bytes.MappedBytes.readPosition(MappedBytes.java:350)
	at net.openhft.chronicle.bytes.MappedBytes.readPosition(MappedBytes.java:53)
	at net.openhft.chronicle.wire.BinaryWire$BinaryValueIn.marshallable(BinaryWire.java:3325)
	at net.openhft.chronicle.wire.Wires.objectMap(Wires.java:444)
	at net.openhft.chronicle.wire.Wires.object0(Wires.java:505)
	at net.openhft.chronicle.wire.ValueIn.object(ValueIn.java:592)
	at software.chronicle.datagrid.internal.queue.CloseableTailer.next(CloseableTailer.java:65)
	at software.chronicle.datagrid.internal.map.QueueObjectSubscription$TopicSubscriberHandler.guardedSingleAction(QueueObjectSubscription.java:353)
	at software.chronicle.datagrid.internal.map.QueueObjectSubscription$AbstractSubscriberHandler.action(QueueObjectSubscription.java:429)
	at net.openhft.chronicle.threads.VanillaEventLoop.runAllMediumHandler(VanillaEventLoop.java:375)
	at net.openhft.chronicle.threads.VanillaEventLoop.runMediumLoopOnly(VanillaEventLoop.java:291)
	at net.openhft.chronicle.threads.VanillaEventLoop.runLoop(VanillaEventLoop.java:254)
	at net.openhft.chronicle.threads.VanillaEventLoop.run(VanillaEventLoop.java:227)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

appender:
```
"main" #1 prio=5 os_prio=31 tid=0x00007ff06680d000 nid=0x1003 runnable [0x0000700009fd0000]
   java.lang.Thread.State: RUNNABLE
	at java.io.RandomAccessFile.setLength(Native Method)
	at net.openhft.chronicle.bytes.MappedFile.acquireByteStore(MappedFile.java:352)
	- locked <0x000000074615ebb0> (a java.lang.Class for java.nio.channels.FileChannel)
	- locked <0x000000074603b310> (a net.openhft.chronicle.bytes.MappedFile)
	at net.openhft.chronicle.bytes.MappedFile.acquireByteStore(MappedFile.java:310)
	at net.openhft.chronicle.bytes.MappedBytes.acquireNextByteStore0(MappedBytes.java:434)
	- locked <0x000000074601df40> (a net.openhft.chronicle.bytes.MappedBytes)
	at net.openhft.chronicle.bytes.MappedBytes.acquireNextByteStore(MappedBytes.java:426)
	at net.openhft.chronicle.bytes.MappedBytes.write(MappedBytes.java:562)
	at net.openhft.chronicle.bytes.MappedBytes.write(MappedBytes.java:53)
	at net.openhft.chronicle.bytes.StreamingDataOutput.write(StreamingDataOutput.java:310)
	at net.openhft.chronicle.wire.BinaryWire$FixedBinaryValueOut.bytes0(BinaryWire.java:1354)
	at net.openhft.chronicle.wire.BinaryWire$FixedBinaryValueOut.bytes(BinaryWire.java:1331)
	at net.openhft.chronicle.wire.WireMarshaller$BytesFieldAccess.getValue(WireMarshaller.java:720)
	at net.openhft.chronicle.wire.WireMarshaller$FieldAccess.write(WireMarshaller.java:476)
	at net.openhft.chronicle.wire.WireMarshaller.writeMarshallable(WireMarshaller.java:166)
	at net.openhft.chronicle.wire.Marshallable.writeMarshallable(Marshallable.java:141)
	at net.openhft.chronicle.wire.BinaryWire$FixedBinaryValueOut.marshallable(BinaryWire.java:1743)
	at net.openhft.chronicle.wire.ValueOut.object(ValueOut.java:505)
	at software.chronicle.datagrid.internal.queue.ChronicleQueueView.publishAndIndex(ChronicleQueueView.java:508)
	at software.chronicle.datagrid.internal.queue.ChronicleQueueView.publish(ChronicleQueueView.java:500)
	at software.chronicle.datagrid.issue.SubscriberThrottlingTest.lambda$reproduceTopicSubscriber$2(SubscriberThrottlingTest.java:131)
	at software.chronicle.datagrid.issue.SubscriberThrottlingTest$$Lambda$268/306980751.run(Unknown Source)
	at software.chronicle.datagrid.issue.SubscriberThrottlingTest.sendMessages(SubscriberThrottlingTest.java:170)
	at software.chronicle.datagrid.issue.SubscriberThrottlingTest.reproduceTopicSubscriber(SubscriberThrottlingTest.java:131)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at software.chronicle.datagrid.ShutdownHooks$1.evaluate(ShutdownHooks.java:71)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)

```

the appender is on this line:

![image](https://user-images.githubusercontent.com/6506056/82916430-efb1a280-9f69-11ea-81b3-d8dfe0de505b.png)

**Comment by RobAustin** [comment]
mappedFile.acquireByteStore() was both synchronized() at a method level and also locked via synchronized (GLOBAL_FILE_LOCK), Hence, tailers who were trying to acquire an earlier ByteStore where being blocked, the fix was to rework the synchronization. To allow consumers who do not require an increase in the raf.setLength() to proceed unhindered.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.20.101](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
